src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -272,6 +275,26 @@ public class CitizenData implements ICitizenData
      */
     private int inactivityTimer = DISABLED;
 
+    /**
+     * The list of available quests the citizen can give out.
+     */
+    private final List<ResourceLocation> availableQuests = new ArrayList<>();
+
+    /**
+     * The list of participating quests the citizen can give out.
+     */
+    private final List<ResourceLocation> participatingQuests = new ArrayList<>();
+
+    /**
+     * Tracking quests the citizen was the quest giver in.
+     */
+    private final List<ResourceLocation> finishedQuests = new ArrayList<>();
+
+    /**
+     * Tracking quests the citizen participated in.
+     */
+    private final List<ResourceLocation> finishedQuestParticipation = new ArrayList<>();
```

Comment:
I want to be able to easily include them in citizen matching

Reply:
oh I wouldnt change that, I'd just do a Map<Citizen, thiese lists here and redirect lookups to the manager when a citizen is queried. So functionally its the very same, the citizens just lookup things in the managers map instead of their own list, just so you don't accidently create mismatched data between the quest manager and the citizen store

Reply:
Yeah, but that makes matching them not possible. At the very least not for finished participation.

Active one we could move, but would go through additional corners of querying from interactions etc.